hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/model/HoodieInternalRow.java

```diff
@@ -94,7 +94,11 @@ private HoodieInternalRow(UTF8String[] metaFields,
 
   @Override
   public int numFields() {
-    return sourceRow.numFields();
+    if (sourceContainsMetaFields) {
```

Comment:
Great catch! Let's add test for it

Reply:
Add UT in TestHoodieInternalRow#testNumFields